modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/AuditBarChart.js

```diff
@@ -25,35 +25,77 @@ import {
 	YAxis,
 } from 'recharts';
 
-const BAR_CHART = {
-	barHeight: 16,
-	dotRadiusMax: 35,
-	dotRadiusMin: 5,
-	fillXAxis: '#666',
-	height: 400,
-	legendHeight: 70,
-	stroke: '#E7E7ED',
-	width: 1150,
-};
+import {BAR_CHART, COLORS} from '../utils/constants';
+
+export default function AuditBarChart({categories, rtl}) {
+	const auditBarChartData = useMemo(() => {
+		const dataKeys = new Set();
+
+		const bars = categories.reduce((acc, category) => {
+			if (!category.children) {
+				return undefined;
+			}
+
+			const newBar = category.children.reduce(
+				(childAcc, {key: dataKey, name}) => {
+					if (dataKeys.has(dataKey)) {
+						return childAcc;
+					}
+
+					dataKeys.add(dataKey);
+
+					return childAcc.concat({dataKey, name});
+				},
+				[]
+			);
+
+			return acc.concat(newBar);
+		}, []);
+
+		const data = categories.map((category) => {
+			if (!category.children) {
+				return category;
+			}
+
+			return category.children.reduce(
+				(acc, {key, value}) => {
+					return {
+						...acc,
+						[key]: value,
+					};
+				},
+				{name: category.name}
+			);
+		});
+
+		return {bars, data};
+	}, [categories]);
+
+	const {bars, data} = auditBarChartData;
+
+	const height =
+		bars === undefined
```

Comment:
Rather than a strict check against `undefined`, would it still be fine if you had `!bars ? ...`? Might help in reducing the number of columns this takes up. Or you could even flip the expression and do `bars ? BAR_CHART.height : ...`

Reply:
Seems `bars` shouldn't be `undefined` because it should return (at least) an empty array (that's assuming the other thing you commented on is fixed). So maybe `bars.length ? ... : ...`